### PR TITLE
Align documentation of connection management with connectivity concepts publication (v2)

### PR DESCRIPTION
**Pull request by jhnnsnk**
This PR replaces #2992, for the discussion refer to the original PR.

### CODE REVIEW

**Review by heplesser — CHANGES_REQUESTED** [review]
@jhnnsnk Thanks! I just have minor suggestions.
@jessica-mitchell Can you review this PR as soon as possible for inclusion on 3.7?
@terhorstd Adding this to the 3.7 milestone, it is only documentation but important documentation.

File: doc/htmldoc/synapses/connectivity_concepts.rst [hunk 1]
```diff
@@ -0,0 +1,586 @@
...
+.. rst-class:: center
+
+    Basic connection rules commonly used in the computational neuroscience community. For more details, go to the section :ref:`conn_rules` or just click on one of the illustrations.
+
+
+This documentation not only describes how to define network connectivity in NEST, but also provides details on connectivity concepts.
+The article "Connectivity concepts in neuronal network modeling" [1]_ serves as a permanent reference for a number of connection rules and we suggest to cite it if rules defined there are used.
+This documentation instead represents a living reference for these rules, and deviations from and extensions to what is described in the article will be highlighted.
+
+.. The same article also introduces a graphical notation for neuronal network diagrams which will be curated in the documentation of :doc:`NEST Desktop <desktop:user/usage-advanced/network-graph>`.
```
[the remark this answers is not in the document]
Comment (jhnnsnk) [reply doc/htmldoc/synapses/connectivity_concepts.rst:57]: This statement was already prepared such that we can get it in quickly once the graphical notation is in NEST Desktop. This is the case now and I will include it.

File: doc/htmldoc/synapses/connection_generator.rst [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+pattern in a library-specific way. The Connection Generator is handed
+to :py:func:`.Connect` under the key ``cg`` of the connection specification
+dictionary and evaluated internally. If the Connection Generator
+provides values for connection weights and delays, their respective
+indices can be specified under the key ``params_map``. Alternatively,
+all synapse parameters can be specified using the synapse
+specification argument to ``Connect()``.
+
+The following listing shows an example for using CSA (`Connection Set Algebra <https://github.com/INCF/csa>`_ [1]_) in NEST via the Connection
+generator interface and randomly connects 10% of the neurons from
```
[the remark this answers is not in the document]
Comment (jhnnsnk) [reply doc/htmldoc/synapses/connection_generator.rst]: Fixed.

**Review by heplesser — APPROVED** [review]

**Comment by jessica-mitchell** [comment]
> @jessica-mitchell Can you review this PR as soon as possible for inclusion on 3.7? 

yep will look at it tonight

**Review by jessica-mitchell — APPROVED** [review]

**Pull request merged**